src/functions/rolePolicyFactory.ts

```diff
@@ -0,0 +1,84 @@
+import { APIGatewayAuthorizerResult, Statement } from "aws-lambda";
+import { Jwt } from "jsonwebtoken";
+import { AccessHttpVerbMap } from "../models/AccessHttpVerbMap";
+import { ILogEvent } from "../models/ILogEvent";
+import { HttpVerb } from "../services/http-verbs";
+import Role, { getLegacyRoles } from "../services/roles";
+import StatementBuilder from "../services/StatementBuilder";
+import newPolicyDocument from "./newPolicyDocument";
+
+type NonEmptyArray<T> = [T, ...T[]];
+
+const accessToHttpVerbs: AccessHttpVerbMap = {
+  read: ["GET", "HEAD"],
+  write: ["*"],
+};
+
+const Configuration: AuthorizerConfig = {
+  CVSFullAccess: ["/*"],
+  CVSPsvTester: ["/*"],
+  CVSHgvTester: ["/*"],
+  CVSAdrTester: ["/*"],
+  CVSTirTester: ["/*"],
+  VTMAdmin: ["/*"],
+  Certs: ["/*/v1/document-retrieval", "/*/v1/document-retrieval/*"],
+  VehicleData: ["/*/v1/enquiry", "/*/v1/enquiry/*"],
+  DVLATrailers: ["/*/trailers", "/*/trailers/*"],
```

Comment:
Do Certs, VehicleData, and DVLATrailers need adding to backwardsCompatibleRoleNames?
![image](https://user-images.githubusercontent.com/13391709/198042111-3435e132-b6e2-4c61-91a7-28fa53840b51.png)

Reply:
That's the existing codebase, so they're not in there today...I guess I add a test and find out?

Reply:
@Ian-Hodges it appears these three roles can't have previously worked as they weren't in the list. Now added them 👍

Reply:
It looks like there are no unit tests for the legacy roles. You could do the lot with the following.
```
it.each([
    ['CVSFullAccess', '/*', 1],
    ['CVSPsvTester', '/*', 1],
    ['CVSHgvTester', '/*', 1],
    ['CVSAdrTester', '/*', 1],
    ['CVSTirTester', '/*', 1],
    ['VTMAdmin', '/*', 1],
    ['Certs', '/v1/document-retrieval/*', 2],
    ['VehicleData', '/v1/enquiry/*', 2],
    ['DVLATrailers', '/v1/trailers/*', 2]])(
    "legacy roles should return valid write statements on valid JWT", async (legacyRolename, arn, statementCount) => {
      const jwtJsonClone = JSON.parse(JSON.stringify(jwtJson));
      jwtJsonClone.payload.roles = [legacyRolename];
      (getValidJwt as jest.Mock) = jest.fn().mockReturnValue(jwtJsonClone);

      const returnValue: APIGatewayAuthorizerResult = await authorizer(event, exampleContext());

      expect(returnValue.principalId).toEqual(jwtJson.payload.sub);

      expect(returnValue.policyDocument.Statement.length).toEqual(statementCount);
      expect(returnValue.policyDocument.Statement).toContainEqual({
        Effect: "Allow",
        Action: "execute-api:Invoke",
        Resource: `arn:aws:execute-api:eu-west-1:*:*/*/*${arn}`,
      });
    });
```

Reply:
The authoriser tests all rely on correctly processing legacy roles

Reply:
Are you sure? The code that uses backwardsCompatibleRoleNames is not covered?
![image](https://user-images.githubusercontent.com/13391709/198070930-be5203a1-502a-48ee-b129-530efffa335b.png)

Reply:
@Ian-Hodges I just added a test that uses `DVLATrailers`, I had to specifically manipulate the array you're talking about to get it to work - so I know it's used in the test I wrote.

The test fakes the JWT to make sure the roles are added, but otherwise performs an end to end call of the authoriser.

As to whether or not every logic path is covered, I'm sure there are gaps as this is the legacy codebase and not the purpose of this PR.

I'm happy to help you work on increasing the overall test coverage if that's something you'd like to help with - but I do need to focus on the tickets we're working on right now 👍

Reply:
To confirm, are the legacy roles in the jwt CVSFullAccess, CVSPsvTester etc... or are they CVSFullAccess.read, CVSFullAccess.write, CVSPsvTester.read, CVSPsvTester.write etc...

Reply:
I think I can see where I've misunderstood things. I thought the backwardsCompatibleRoleNames logic was part of your work to support legacy roles. It is in fact a previous developer doing the same thing when CVS switched roles from a `name ` format to a `name.access` format. I suspect it's redundant now and can be removed, but that's for another day.